src/ConstructiveSolidGeometry/SurfacePrimitives/ConalPlane.jl

```diff
@@ -27,11 +27,12 @@ get_z_limits(c::ConalPlane{T}) where {T} = (_left_linear_interval(c.z), _right_l
 in(p::AbstractCoordinatePoint, c::ConalPlane) =
     _in_z(p, c.z) && _eq_φ(p, c.φ) && _in_cyl_r(p, get_r_at_z(c, p.z))
 
-function sample(c::ConalPlane{T}, step::Quantity{<:Real, Unitful.𝐋}) where {T}
+#function sample(c::ConalPlane{T}, step::Quantity{<:Real, Unitful.𝐋}) where {T}
+function sample(c::ConalPlane{T}, step::Real) where {T}
```

Comment:
That is code duplication.
Just calculate the 3 number of samples per dimension based on `step`
and call the other method.

Reply:
It is not code duplication. In one case you cave a constant step regardless of r width, which does not translate into a constant number of samples. You cannot know the number of samples output by `sample(c::AbstractSurfacePrimitive, step::Real)` without knowing the geometry, whereas `sample(c::AbstractSurfacePrimitive, (a,b,c) )` will always output a*b, a*c, or b*c samples regardless of the geometry.

Reply:
Also the `NTuple{3,Int}` method is meant to always contain the edges of the primitive whereas the the `step::Real` method is not

Reply:
I count 5 copied lines. 
And here you also only need 2 intergers: `NTuple{2,Int}`
You do now the geometry. 
```julia
sample(c::AbstractSurfacePrimitive, step::Real) = 
   sample(c, (phi_width(c)/step, z_width(c)/step)) # Pseudo code
```
The edges should always be included.

Reply:
Yes you only need two integers in all of these but then you lose the interpretability of reading `NTuple{2,Int}` as in a certain coordinate system. This would then lead to loss of generality of the sample_surface function since you would have to map `NTuple{3,Int}` to different 'NTuple{2,Int}'
Yes you know the geometry, I was just making the point of how these sample functions differ.
In your pseudocode `z_width(c)` varies with r and therefore not a constant parameter you can just pass

Reply:
I agree that edges should be included. Therefore the sample(Nsamps) will be the default fast and easy sampling method. However I still see value in having a base sampling function which samples the surface with an even step. There is no loss of performance or code duplication here although yes the code is very similar, they differ in a small but meaningful way.

Reply:
This goes along with my comment on the `Cone{T}` sampling:

I see David's point. Let us assume that we want to sample a `Cone{T}` that is 0.1m wide at the bottom and 0.05m wide at the top with a `step` of 0.01m. According to the pseudo code, this would call `sample` with 10 samples in `r`. However, these 10 samples will be differently spaced at different height and not fulfill the requirement of all being 0.01m apart.

![Cone_spacing](https://user-images.githubusercontent.com/30291312/106942460-7c7d4480-6724-11eb-8b03-a026d48e4808.png)

Reply:
See my comment below the other comment of @fhagemann. 
So okya, leave the two methods like they are but only some minor changes. 
Please add the keyword `stop` to the `range`-methods to increase readability. 
Also, create the ranges befor the loop and calculate their length and allocate the respective vector of points
such that you do not need to use `push!` anymore. 
Actually, that is how we can get rid of the code duplications. Create one `method` where you parse two ranges/vectors to it.

Reply:
I mark this as resolved and continue to write to the conversation below as it is basically the same discussion.